### ISSUE TITLE
tacotron2 vs fastspeech2 speech quality issue

### ISSUE THREAD

**Issue by sciai-ai** [espnet/espnet#3302]
I have trained both tacotron2 (200 epochs) and fastspeech 2 (1000 epochs) models on my dataset (fs=22050) with PWG vocoder. The tacotron2 synthesised speech sounds very good but the fastspeech2 sounds noisy. I have attached the samples. I need some help to understand why the fastspeech2 is producing noisy speech and how can  I improve it.

[taco.wav 2.zip](https://github.com/espnet/espnet/files/6679890/taco.wav.2.zip)
[taco.wav.zip](https://github.com/espnet/espnet/files/6679891/taco.wav.zip)

**Comment by kan-bayashi** [comment]
It seems the samples are the same.
Could you paste your config or reproducible command?

**Comment by sciai-ai** [comment]
here is the sample from fast speech, you can notice some noise in speech
[fastspeech2.wav.zip](https://github.com/espnet/espnet/files/6683449/fastspeech2.wav.zip)

**Comment by kan-bayashi** [comment]
Could you give me more detail? (config, training curve and so on)
How was the sample of Griffin Lim? I'm wondering where the noise came from (seq2seq or vocoder).

**Comment by sciai-ai** [comment]
I also wonder whether is vocoder or seq2seq, I used the same PWG vocoder which was used to synthesize for tacotron2. and this vocoder was trained on the same data used for training tacotron2.

Here is more detail files, let mw know if you need more info:

[images_fs2.zip](https://github.com/espnet/espnet/files/6683747/images_fs2.zip)

[config.yaml.zip](https://github.com/espnet/espnet/files/6683751/config.yaml.zip)
[sample_GL.wav.zip](https://github.com/espnet/espnet/files/6683828/sample_GL.wav.zip)

**Comment by kan-bayashi** [comment]
It seems seq2seq model is no problem because GL sample is fine.
Your fastspeech2 sample with vocoder looks something wrong (FS2 GL vs. FS2 vocoder vs. Taco2 vocoder).
<img width="1292" alt="スクリーンショット 2021-06-21 12 29 08" src="https://user-images.githubusercontent.com/22779813/122703483-aa56c600-d28c-11eb-9a69-f4827b106aa9.png">
I'm not sure why this is happened. Could you re-check the vocoder application part is correct?

**Comment by sciai-ai** [comment]
Seems some clipping in FS2 spectrogram. 
How can i check for vocoder: norm or denorm features?

**Comment by kan-bayashi** [comment]
In the case of GL, the clipping is normal since the fbank is limit to 80-7600 Hz.
But when applying neural vocoder, the components of all frequency range should be recovered.
In the case of Taco2, all frequency components are recovered but in the caes of FS2, sometimes recovered and sometimes missing. This is very strange.

Several questions:
- Is the training data the same among vocoder, tacotron2, and fastspeech2?
- Do you use different statistics for vocoder and fastspeech2?

**Comment by sciai-ai** [comment]
I am now training fastspeech2 model again to confirm. I noticed that the fs2  training step 7 expects train.ave.loss.pth  from tacotron2 model but after training tacotron2 I only have train.best.loss.pth. I just renamed the file and make the code run. I hope thats not the issue?

1. training data is same
2. How can i confirm the statistics? The same vocoder works well for tacotron2 but produce noise/clipping as you observed in audacity.

**Comment by kan-bayashi** [comment]
> How can i confirm the statistics? The same vocoder works well for tacotron2 but produce noise/clipping as you observed in audacity.

You can try case 2 in the url:
https://github.com/kan-bayashi/ParallelWaveGAN#decoding-with-espnet-tts-models-features

**Comment by sciai-ai** [comment]
The FS2 synthesis quality is good after training from scratch, I think could be possibly some slight data mismtach between tacotron and FS2, like adding pause at start and end of sentences using MFA.

Thanks @kan-bayashi
